src/components/Common/UI/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,102 @@
+import type { Key } from 'react-aria-components'
+import {
+  ComboBox as AriaComboBox,
+  Button,
+  Input,
+  ListBox,
+  ListBoxItem,
+  Popover,
+} from 'react-aria-components'
+import { useTranslation } from 'react-i18next'
+import type { FocusEvent, InputHTMLAttributes } from 'react'
+import { useId } from 'react'
+import classNames from 'classnames'
+import type { FieldLayoutProps } from '../FieldLayout'
+import { FieldLayout } from '../FieldLayout'
+import styles from './ComboBox.module.scss'
+import { useTheme } from '@/contexts/ThemeProvider'
+import CaretDown from '@/assets/icons/caret-down.svg?react'
+export interface ComboBoxOption {
+  label: string
+  value: string
+}
+
+export interface ComboBoxProps
+  extends FieldLayoutProps,
+    Pick<InputHTMLAttributes<HTMLInputElement>, 'className' | 'id' | 'name' | 'placeholder'> {
+  isDisabled?: boolean
+  isInvalid?: boolean
+  label: string
+  onChange?: (value: string) => void
+  onBlur?: (e: FocusEvent) => void
+  options: ComboBoxOption[]
+  value?: string
+}
+
+export const ComboBox = ({
+  className,
+  description,
+  errorMessage,
+  id: providedId,
+  isDisabled,
+  isInvalid,
+  isRequired,
+  label,
+  onChange,
+  onBlur,
+  options,
+  placeholder,
+  value,
+  ...props
+}: ComboBoxProps) => {
+  const { t } = useTranslation()
+  const generatedInputId = useId()
+  const inputId = providedId || generatedInputId
+  const generatedErrorMessageId = useId()
```

Comment:
If you rebase you'll now be able to update this to use `useFieldIds` util instead! see other form components in the UI directory for reference

Reply:
Updated - WIll be in next revision